packages/design-system/src/utils/typed-event-bus.ts

```diff
@@ -0,0 +1,36 @@
+import type { UnregisterFn } from './event-bus';
+import { createEventBus as createUntypedEventBus } from './event-bus';
+
+export type Listener<Payload> = (payload: Payload) => void;
+
+export type Payloads<ListenerMap> = {
+	[E in keyof ListenerMap]: unknown;
+};
+
+export interface TypedEventBus<ListenerMap extends Payloads<ListenerMap>> {
+	on: <EventName extends keyof ListenerMap & string>(
+		eventName: EventName,
+		fn: Listener<ListenerMap[EventName]>,
+	) => UnregisterFn;
+
+	once: <EventName extends keyof ListenerMap & string>(
+		eventName: EventName,
+		fn: Listener<ListenerMap[EventName]>,
+	) => UnregisterFn;
+
+	off: <EventName extends keyof ListenerMap & string>(
+		eventName: EventName,
+		fn: Listener<ListenerMap[EventName]>,
+	) => void;
+
+	emit: <EventName extends keyof ListenerMap & string>(
+		eventName: EventName,
+		event?: ListenerMap[EventName],
+	) => void;
+}
+
+export function createTypedEventBus<
+	ListenerMap extends Payloads<ListenerMap>,
+>(): TypedEventBus<ListenerMap> {
+	return createUntypedEventBus() as TypedEventBus<ListenerMap>;
+}
```

Comment:
Great to see you've added event bus types! 

I think we should merge the two event bus types and just set the default generic to `Record<string, unknown>` when creating an event bus. I don't see a reason to keep the two create functions separate.

Reply:
Makes sense. I kept them separate to have a clear refactoring path to the new one, but we can also combine them already at this point. However, I can't use `Record<string, unknown>` as that will break most of the usage sites because they are missing the correct event types. Will type it with `Record<string, any>` for now.